kubernetes/client/apis/custom_objects_api.py

```diff
@@ -1656,8 +1657,8 @@ def patch_cluster_custom_object_with_http_info(self, group, version, plural, nam
             select_header_accept(['application/json'])
 
         # HTTP header `Content-Type`
-        header_params['Content-Type'] = self.api_client.\
-            select_header_content_type(['application/merge-patch+json'])
+        header_params['Content-Type'] = params.get('content_type', self.api_client.\
+            select_header_content_type(['application/merge-patch+json']), 'PATCH', body_params)
```

Comment:
This all started due to that we added 'application/json-patch+json' for custom objects, so should that not be part of the list of available content types?

Reply:
Also assuming this is an example on the changes to all the patch methods, since there are several more in this file.

Reply:
Yes, should be but this PR uses current code from master branch (https://github.com/kubernetes-client/python/blob/master/kubernetes/client/apis/custom_objects_api.py#L1659). AFAIK @roycaihw removed this option when the problem appeared.

Reply:
Yes, I use this PR to discuss final solution and then I will change the openapi-generator to add this to all methods.

Reply:
I removed 'application/json-patch+json' from custom objects because the existing select_header_content_type couldn't support it without breaking old clients. With this change from @tomplus, we should be able to add 'application/json-patch+json' for custom objects